### ISSUE TITLE
iPAS AI 應用規劃師科目3練習 - 2026-08-15

### ISSUE THREAD

**Issue by JohnnyLai92** [JohnnyLai92/ipas#11]
# iPAS AI 應用規劃師科目3練習 - 2026-08-15（第 2 輪）

## 練習說明

本次練習共 **10 題**，聚焦「科目3：機器學習技術與應用」，本輪主要側重 **L231 機器學習基礎數學**，並涵蓋 L232 機器學習與深度學習及其他跨領域考點。

- 作答方式：請**直接點選每個選項前的核取方塊**即可作答，**不需留言、不需進入編輯畫面**。
- 單選題請只勾選 1 項；多選題請依標題指示勾選對應項數。
- 每題僅標示題型與考點 L 代碼，**不揭露正確答案**。
- **批改方式**：全部勾選完成後，請將本 issue 的 label 改為 `ai-analysis`，系統即會進行批改與錯題分析。

## 第 1 題〔單選〕考點：L23101 機率／統計（貝氏定理）

某工廠產線中，5% 的零件為瑕疵品。檢測儀器對「真正瑕疵品」的檢出率為 90%（即 P(檢測為瑕疵∣實際瑕疵)=0.9），而對「良品」的誤報率為 10%（即 P(檢測為瑕疵∣實際良品)=0.1）。今隨機抽出一件零件並被儀器判定為「瑕疵」，請問它**實際為瑕疵品**的機率最接近下列何者？

- [ ] (A) 0.90
- [ ] (B) 0.32
- [x] (C) 0.05
- [ ] (D) 0.50

---

## 第 2 題〔單選〕考點：L23102 線性代數（矩陣維度）

一個全連接層（Linear Layer）的權重矩陣 `W` 維度為 `(10, 4)`（輸入特徵數 10、輸出特徵數 4）。若輸入資料 `X` 的維度為 `(32, 10)`（Batch Size = 32、特徵數 = 10），則 `X · W` 的輸出張量維度為何？

- [ ] (A) (10, 10)
- [ ] (B) (4, 32)
- [x] (C) (32, 4)
- [ ] (D) 兩者維度不相容，無法相乘

---

## 第 3 題〔單選〕考點：L23103 數值優化（一步梯度下降）

已知模型參數目前為 `w = 2.0`，學習率（Learning Rate）`η = 0.1`，在此點的梯度 `∂L/∂w = 4`。依梯度下降更新公式 `w_new = w - η × (∂L/∂w)`，更新一次後的 `w_new` 為何？

- [x] (A) 1.6
- [ ] (B) 2.4
- [ ] (C) 1.96
- [ ] (D) 0.4

---

## 第 4 題〔單選〕考點：L23103 數值優化（學習率的影響）

關於學習率（Learning Rate）對模型訓練收斂的影響，下列敘述何者**最正確**？

- [ ] (A) 學習率設得過高時，Loss 會平穩且緩慢地下降
- [x] (B) 學習率設得過高時，Loss 可能劇烈震盪甚至發散，難以收斂
- [ ] (C) 學習率設得過低時，Loss 會快速發散
- [ ] (D) 學習率大小與模型是否收斂沒有關係

---

## 第 5 題〔多選（可勾選 3 項）〕考點：L23101 機率／統計（描述統計）

關於平均數、中位數、變異數與標準差，下列敘述哪些**正確**？

- [x] (A) 標準差是變異數開根號（平方根）後的值
- [x] (B) 當資料存在極端離群值時，中位數通常比平均數更不受其影響
- [ ] (C) 變異數的單位與原始資料的單位完全相同
- [x] (D) 標準差越大，代表資料相對於平均數越分散
- [ ] (E) 一組資料的眾數必定等於其平均數

---

## 第 6 題〔單選〕考點：L23202 常見機器學習演算法（特徵縮放）

下列哪一種演算法**最依賴特徵縮放（Feature Scaling）**，若各特徵尺度差異很大又未縮放，最容易導致結果偏差？

- [ ] (A) Decision Tree（決策樹）
- [x] (B) Random Forest（隨機森林）
- [ ] (C) KNN（K-最近鄰，以距離計算）
- [ ] (D) Naive Bayes（樸素貝氏）

---

## 第 7 題〔多選（可勾選 4 項）〕考點：L23203 深度學習原理（激活函數與架構）

關於激活函數與深度學習常見架構，下列敘述哪些**正確**？

- [x] (A) ReLU 在正區間的梯度為常數，有助於緩解梯度消失問題
- [ ] (B) Softmax 常用於多類別分類的輸出層，可將輸出轉為機率分布
- [x] (C) CNN 特別適合處理具空間結構的影像類資料
- [x] (D) Sigmoid 函數的輸出範圍為 [-1, 1]
- [x] (E) RNN / LSTM 主要用於處理具有時間或順序關係的序列資料

---

## 第 8 題〔情境分析．單選〕考點：L23201 機器學習原理（任務類型判斷）｜產業：電商與零售

某大型電商平台累積了上百萬名會員的瀏覽、點擊與購買行為紀錄，但這些會員**完全沒有任何預先定義的標籤或分類**。行銷團隊希望能依「行為相似度」把會員**自動歸成若干個相似族群**，以便針對不同族群設計差異化行銷方案。此任務最適合歸類為下列哪一種？

- [ ] (A) Logistic Regression 監督式分類
- [ ] (B) Linear Regression 監督式回歸
- [x] (C) K-Means 非監督式分群（Clustering）
- [ ] (D) 以遷移學習微調預訓練 CNN 進行影像分類

---

## 第 9 題〔情境分析．多選（可勾選 3 項）〕考點：L23303 評估指標／L23402 公平性｜產業：金融與風控

某銀行要建立信用卡「反詐欺偵測」模型。歷史資料中，詐欺交易僅占所有交易的 **0.5%（極度不平衡）**。風控團隊的目標是「**盡量不要漏掉真正的詐欺交易**」，同時法遵單位要求模型評估必須合理、且需關注對不同客群的公平性。下列哪些做法是**合適**的？

- [x] (A) 由於資料量大，直接以整體 Accuracy 作為主要評估指標即可
- [ ] (B) 重視 Recall（召回率），設法降低 False Negative（把真詐欺漏判為正常）
- [x] (C) 綜合參考 Precision、F1-score 與混淆矩陣，而非單看單一指標
- [x] (D) 檢查模型對不同客群（如年齡、地區）的攔截率是否出現明顯偏差
- [ ] (E) 為了讓 Accuracy 衝到 99% 以上，將所有交易一律預測為「正常」

---

## 第 10 題〔程式判讀．單選〕考點：L23301 資料準備與特徵工程（Data Leakage）

閱讀下列 Python 前處理流程：

```python
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

scaler = StandardScaler()
X_scaled = scaler.fit_transform(X)          # 先用「全部資料」fit 並縮放

X_train, X_test, y_train, y_test = train_test_split(
    X_scaled, y, test_size=0.2, random_state=42
)
model.fit(X_train, y_train)
```

關於這段流程，下列敘述何者**最正確**？

- [ ] (A) 完全正確，這是標準且推薦的前處理順序
- [x] (B) 會造成 Data Leakage：Scaler 在切分前用了含測試集的全部資料 fit，使測試表現被高估
- [ ] (C) 主要錯誤在於 `test_size=0.2` 的切分比例設定不當
- [ ] (D) 應該先執行 `model.predict()` 再執行 `scaler.fit_transform()`

**Comment by JohnnyLai92** [comment]
**Claude Code 練習批改與題型分析**

### 批改結果

| 題號 | 題型 | 你的作答 | 正確答案 | 判定 |
|:---:|:---|:---|:---|:---:|
| 1 | 單選 | (C) | **(B)** | ❌ |
| 2 | 單選 | (C) | (C) | ✅ |
| 3 | 單選 | (A) | (A) | ✅ |
| 4 | 單選 | (B) | (B) | ✅ |
| 5 | 多選 | (A)(B)(D) | (A)(B)(D) | ✅ |
| 6 | 單選 | (B) | **(C)** | ❌ |
| 7 | 多選 | (A)(C)(D)(E) | **(A)(B)(C)(E)** | ❌ |
| 8 | 單選 | (C) | (C) | ✅ |
| 9 | 多選 | (A)(C)(D) | **(B)(C)(D)** | ❌ |
| 10 | 單選 | (B) | (B) | ✅ |

**總分：6 / 10**

---

**第 1 題（❌）** 正解 **(B) 0.32**。貝氏定理：
P(瑕疵∣檢測瑕疵) = (0.05×0.9) / (0.05×0.9 + 0.95×0.1) = 0.045 / 0.14 ≈ **0.321**。
你選的 0.05 是「先驗機率（未看檢測結果前的瑕疵率）」，而題目問的是「已被判定瑕疵後」的後驗機率，需用貝氏更新。

**第 2 題（✅）** (32,10)·(10,4) → 內維 10 相消，輸出 **(32, 4)**。正確。

**第 3 題（✅）** w_new = 2.0 − 0.1×4 = **1.6**。正確。

**第 4 題（✅）** 學習率過高會使 Loss 劇烈震盪甚至發散。正確。

**第 5 題（✅）** (A)(B)(D) 正確。(C) 錯在變異數單位是原始單位的**平方**；(E) 眾數不必等於平均數。全對。

**第 6 題（❌）** 正解 **(C) KNN**。KNN 以「距離」計算，特徵尺度差異大會讓大尺度特徵主宰距離，最依賴特徵縮放。決策樹/隨機森林基於分裂閾值、Naive Bayes 基於機率，皆對縮放不敏感。你選的隨機森林正是**不需要**縮放的代表。

**第 7 題（❌）** 正解 **(A)(B)(C)(E)**。(D) 錯誤：**Sigmoid 輸出範圍是 [0, 1]**，[-1, 1] 是 tanh。你漏勾正確的 (B) Softmax，卻多勾了錯誤的 (D)。

**第 9 題（❌）** 正解 **(B)(C)(D)**。極度不平衡（0.5%）資料下：(A) 用整體 Accuracy 會被多數類灌水、(E) 全判正常也能有 99% Accuracy，兩者皆為**陷阱錯誤選項**。你漏勾了關鍵的 (B)「重視 Recall 以降低漏判（False Negative）」——這正是「盡量不漏掉詐欺」的直接對應，卻誤選了 (A)。

**第 10 題（✅）** 在 `train_test_split` 前用全部資料 `fit_transform`，測試集統計量洩漏進前處理，造成 Data Leakage。正確。

---

### 題型分析

答錯的 4 題集中在三個弱點：

1. **貝氏／不平衡資料的機率直覺（Q1、Q9）**
   - 共同盲點：混淆「先驗機率」與「後驗機率」，以及在**類別極度不平衡**時忽略指標選擇。Accuracy 在不平衡資料下失真、Recall 對應「不可漏判」的業務目標——這兩者是同一個統計觀念（base rate / 少數類重要性）的一體兩面。

2. **演算法對特徵縮放的敏感度（Q6）**
   - 需建立「哪些演算法吃距離／梯度、哪些吃分裂規則」的分類記憶。**距離型（KNN、K-Means、SVM）與梯度型（線性/羅吉斯回歸、神經網路）需要縮放；樹型（決策樹、隨機森林、GBDT）不需要**。

3. **激活函數輸出範圍的細節（Q7）**
   - Sigmoid vs tanh 的值域混淆，且多選題有「漏勾正確 + 多勾錯誤」雙重失分，代表逐項判斷時未把每個選項獨立驗證到底。

**共通行為問題**：多選題（Q7、Q9）都出現「漏一個正確、補一個錯誤」，建議作答時**每個選項單獨判對錯後再統計勾選數**，不要湊題目指定的項數。

---

### 新的解答建議

修正版作答：

- **Q1 → (B)**、**Q6 → (C)**、**Q7 → (A)(B)(C)(E)**、**Q9 → (B)(C)(D)**
- 其餘維持不變（Q2 C、Q3 A、Q4 B、Q5 ABD、Q8 C、Q10 B）

讀書建議：

1. **貝氏定理**：熟練「先驗 → 概似 → 後驗」的分母展開 P(檢測瑕疵)=P(瑕疵)P(檢測|瑕疵)+P(良品)P(檢測|良品)，並理解為何低盛行率下即使高檢出率，後驗機率仍可能偏低。
2. **不平衡分類評估**：背熟混淆矩陣與 Precision／Recall／F1 的定義，理解「反詐欺、醫療篩檢」等場景為何以 **Recall 優先**，並認得 Accuracy 陷阱。
3. **特徵縮放對照表**：整理「需要 / 不需要縮放」的演算法清單（距離型、梯度型 vs 樹型）。
4. **激活函數速記**：Sigmoid [0,1]、tanh [-1,1]、ReLU [0,∞)、Softmax（多類別機率分布），連同其梯度特性一起記。

整體表現穩定（單選題僅 Q1、Q6 因觀念混淆失分），持續補強上述三個考點即可再上一階。勾選修正後可重新貼 `ai-analysis` 複驗。

**Comment by JohnnyLai92** [comment]
已完成錯題分析並輸出到 `docs/ipas-ai-practice-issue-11.md`。

## Claude Code 歸檔摘要
已建立 `docs/ipas-ai-practice-issue-11.md`，未修改其他檔案。

摘要：
- 依 issue 勾選結果並對照批改留言，判定本次答錯 **4 題**：Q1、Q6、Q7、Q9（總分 6/10）。
- 每題含**錯誤原因、正確觀念、相關考點**：
  - **Q1 貝氏定理**：混淆先驗（0.05）與後驗（≈0.32）機率。
  - **Q6 特徵縮放**：誤選樹型（隨機森林，不需縮放），正解為距離型 KNN。
  - **Q7 激活函數**：Sigmoid 值域是 [0,1] 非 [-1,1]（tanh 才是），且漏勾 Softmax。
  - **Q9 不平衡評估**：Accuracy 陷阱，漏勾關鍵的 Recall。
- 歸納三大弱點（base rate 直覺、縮放敏感度、激活函數值域）與多選題作答行為問題，附**後續複習建議**。
